qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -764,3 +765,30 @@ def writeSettings(self, settings):
     def override_python_input(self):
         """Replace python input with a call to a qinputdialog"""
         builtins.input = QAppThreadCall(input_qinputdialog)
+
+    def attempt_to_restore_state(self, state):
+        success = self.restoreState(state, SAVE_STATE_VERSION)
+        if success:
+            return
+
+        # The version number of the supplied state is older than the current version
+        reply = QMessageBox.question(
+            self,
+            "Layout Restoration",
+            "The selected layout is incompatible with this version of Workbench. Workbench will attempt to restore \\"
+            "the layout, but it may appear differently from before.\nDo you wish to continue?",
+            QMessageBox.Yes | QMessageBox.No)
+        if not reply == QMessageBox.Yes:
+            return
+
+        for version in range(0, SAVE_STATE_VERSION):
+            success = self.restoreState(state, version)
+            if success:
```

Comment:
same as above, can just have:
```suggestion
            if self.restoreState(state, version):
```